main/annotations/injectable.ts

```diff
@@ -15,8 +15,8 @@ import { getRootInjector } from '../core/util.functions';
  *
  * @param configuration
  */
-export function Injectable(configuration?: IInjectionConfiguration) {
-    return (type: any) => {
+export function Injectable<T>(configuration?: IInjectionConfiguration<T>) {
```

Comment:
so `IInjectionConfiguration` already accepts a type and calculates the constructor params for it I assume